### ISSUE TITLE
Index Select on Particles

### ISSUE THREAD

**Issue by brugger1** [visit-dav/live-customer-response#14]
From: Allen Sanderson <allen@sci.utah.edu> 
Sent: Friday, March 13, 2020 11:12 AM
To: Tony Saad <tony.saad@chemeng.utah.edu>
Cc: Biagas, Kathleen <biagas2@llnl.gov>; Maguire, Alister Owen <maguire7@llnl.gov>
Subject: Re: Index Select on Particles

Hi Tony,

I just tried using the index select on some particle data and while it does do the index select the scalar values are lost and the particles are black. I have attached the original and with the index select. The legend is the same but the values go from 0 to 1. Which tells me the point/cell data is not getting on to the points.

I have not looked at this filter in quite sometime. I am cc’ing two of the VisIt developers to see if they have any immediate ideas of what may be the issue. Otherwise a bug report will need to be submitted.

Cheers,

Allen

Allen Sanderson
SCI Institute
University of Utah

On Mar 12, 2020, at 9:41 PM, Tony Saad <tony.saad@chemeng.utah.edu> wrote:

Hi Allen,

In the past, I used to be able to add an "Index Select" operator to a particle plot so that I can skip every two particles for example...
I can't seem to do that anymore.
Any ideas?

-- 
Tony Saad
Assistant Professor of Chemical Engineering
University of Utah
50 S Central Campus Drive, MEB 2286
Salt Lake City, Utah 84112
Tel: 801-585-0344
www.tonysaad.net
  
![indexselect_bug1](https://user-images.githubusercontent.com/1102718/76651405-b6989300-6521-11ea-914a-52acab2666f2.png)

![indexselect_bug2](https://user-images.githubusercontent.com/1102718/76651438-c912cc80-6521-11ea-87ef-c5305ec3f00a.png)

**Comment by brugger1** [comment]
I'm not familiar with that issue. I'll look into it. It will probably be Monday.

**Comment by markcmiller86** [comment]
I will try to take a quick look

**Comment by markcmiller86** [comment]
@ARSanderson I am unable to reproduce with 3.0.1. Here is what I tried...

1. Open `noise2d.silo` (one of our silo_hdf5_test_data files)
2. PC plot `PointVar`
3. Add index select, set to 1D and incr to 2.
4. Draw. It looks good. No all black 
5. Try switching point type to sphere. It still works

Honestly what it looks like in @ARSanderson example above is that the point density is so high, maybe there is some other odd effect going on. If point density is reduced substantially, does the all-black behavior go away?

![Untitled](https://user-images.githubusercontent.com/5720676/76652277-7c2ff580-6523-11ea-9d8c-b83de446adb4.png)

**Comment by brugger1** [comment]
What version of VisIt are you using and what did you use for the point glyph type? It looks like it was run on a Mac.

**Comment by markcmiller86** [comment]
> What version of VisIt are you using and what did you use for the point glyph type? It looks like it was run on a Mac.

Is this a question for @markcmiller86 or @ARSanderson?

**Comment by brugger1** [comment]
For Allen. You stated you were using 3.0.1 and from the image I can see it was running on a mac.

**Comment by ARSanderson** [comment]
I am running on 3.1 on OS X. The original image was with point glyphs. 

I just switched to spheres and got the following image. The square in the middle is perspective bounding box. When I have seen gooblygook like this in the past we figured out that double arrays were being stuffed into float arrays or visa-versa. This issue has nothing to do with the index select though I though it got fixed.




Cheers,

Allen

**Comment by ARSanderson** [comment]
I figured out the issue. From: https://vtk.org/doc/nightly/html/classvtkMaskPoints.html

vtkMaskPoints <https://vtk.org/doc/nightly/html/classvtkMaskPoints.html> is a filter that passes through points and point attributes from input dataset. (Other geometry is not passed through.) 

Emphasis added. This issue is the same as what we discovered within VisIt proper, that is when it comes to a mesh that is a VTK_VERTEX, the data can be either point (node) or cell (zone) centered. It does not matter. Regardless both point and cell attributes should be passed through. The vtkMaskPoints is not passing through cell data. That is the issue and is a VTK bug. 

Given that I have fix for Uintah reader I am going to change vertex data to node center to get around this issue.

Cheers,

Allen


Allen Sanderson
SCI Institute
University of Utah
